src/shared/lib/AnnotationUtils.ts

```diff
@@ -45,18 +50,113 @@ export function geneToMyCancerGenome(myCancerGenomes:IMyCancerGenome[]):IMyCance
     return map;
 }
 
-export function geneAndProteinPosToHotspots(hotspots:HotspotMutation[]):{[key:string]: boolean}
+export function indexHotspots(hotspots:HotspotMutation[], defaultHotspotType?: string): IHotspotIndex
 {
-    // key: geneSymbol_proteinPosition
-    // (protienPosition: start[_end])
-    const map: {[key:string]: boolean} = {};
+    const index:IHotspotIndex = {};
 
-    // create a map for a faster lookup
-    hotspots.forEach(function(hotspot:HotspotMutation) {
-        const positions = hotspot.residue.match(/[0-9]+/g) || []; // start (and optionally end) positions
-        const key = [hotspot.hugoSymbol.toUpperCase()].concat(positions).join("_");
-        map[key] = true;
+    hotspots.forEach((hotspot:HotspotMutation) => {
+        let geneIndex = index[hotspot.hugoSymbol.toUpperCase()];
```

Comment:
definitely worth saving `hotspot.hugoSymbol.toUpperCase()` in a variable, and also `index[..]`